Lesson4 astafev

astafex opened the pull request:
Домашнее задание 4 на проверку.
Выполнена задание на 5+ (добавление нового номера производится для примера в тестах).

daria-morgen committed:
Добавлен тест для урока 3

RodinDenis committed:
Merge pull request #2 from RodinDenis/add-docs

добавлены ссылки на документацию

RodinDenis committed:
замена импорта Assertions

daria-morgen committed:
Изменен тест для урока 3

daria-morgen committed:
Изменен тест для урока 4.

daria-morgen committed:
Добавлен тест для урока 4 на проверку корректного заполнения коллекции.

daria-morgen committed:
Исправлен тест 4го урока.

daria-morgen committed:
Форматирование.

daria-morgen committed:
Скорректирован тест.

astafex committed:
Merge remote-tracking branch 'origin/lesson-3' into lesson3-astafex

astafex committed:
lesson3.
add interface AccountRepository,
add class AccountService
add class Account

astafex committed:
выполнение домашнего задания 3

astafex committed:
выполнение домашнего задания 3

astafex committed:
Merge pull request #5 from astafex/lesson3-astafex

Lesson3 astafex

astafex committed:
Merge remote-tracking branch 'origin/lesson-4' into lesson4-astafev

astafex committed:
Выполнена часть ДЗ4:
1. Классом AccountRepositoryImpl реализован интерфейс AccountRepository, реализован метод getAllAccountsByClientId
2. В AccountRepositoryImpl добавлены методы для внутреннего применения: parseJsonFile, readFile

astafex committed:
ДЗ4

RodinDenis commented on src/main/java/com/sbrf/reboot/Main.java:
```diff
@@ -1,6 +1,12 @@
 public class Main {
-    public static void main(String[] args) {
+    public static void main(String[] args) throws IOException {
+        AccountRepositoryImpl accountRepository = new AccountRepositoryImpl("src/main/resources/Accounts.txt");
```
рекомендую избегать суффикса Impl. Он "не из Java мира".

RodinDenis commented on src/main/java/com/sbrf/reboot/repository/AccountRepositoryImpl.java:
```diff
@@ -0,0 +1,113 @@
+    public boolean updateClientNumber(long clientId, String currentNumber, String updateNumber) throws IOException {
+        String fileContents = readFile();
+        StringBuilder sb = new StringBuilder(fileContents);
+        Pattern pattern = Pattern.compile(REGEX);
```
лучше компилировать при инициализации. Это дорогая операция

astafex committed:
Обновлено:
1. Переименован класс 'AccountRepositoryImpl' в 'JsonAccountRepository'
2. В классе 'JsonAccountRepository' компиляция PATTERN вынесена в класса

astafex commented:
По замечаниям исправлен код, произведен Merge в develop
